datadog_checks_dev/datadog_checks/dev/plugin/pytest.py

```diff
@@ -38,6 +48,15 @@ def dd_environment_runner(request):
     # Do nothing if no e2e action is triggered and continue with tests
     if not testing_plugin and not e2e_active():  # no cov
         return
+    # If e2e tests are being run it means the environment has
+    # already been spun up so we prevent another invocation
+    elif e2e_testing():  # no cov
+        # Since the scope is `session` there should only ever be one definition
+        fixture_def = request._fixturemanager._arg2fixturedefs[E2E_FIXTURE_NAME][0]
```

Comment:
Ouch. Is there a way not to do that? That looks super fragile.

Reply:
I really tried, but I found absolutely no other way